R/qview.R

```diff
@@ -135,26 +157,37 @@ queueView <- function(
         shiny::need(nrow(.rv$queue_data) > 0, "There are no jobs in the queue for the current user.")
       )
 
+      # cancel_button_creator <- function(index, .ns) {
       cancel_button_column <- purrr::map_chr(
         seq_len(nrow(.rv$queue_data)),
         cancel_button_creator,
         .ns = ns
       )
 
+      #any_button_creator <- function(.button_id, .button_label, index, .ns) {
+      other_button_columns <-
+        purrr::map2(button_ids, button_labels, .f = function(.x, .y) {
+          purrr::map_chr(
+            seq_len(nrow(.rv$queue_data)),
+            any_button_creator,
+            # index = .
+            .button_id = .x,
+            .button_label = .y
+      )}) %>% purrr::set_names(button_labels)
+
       .rv$queue_data %>%
         dplyr::mutate(cancel_button = cancel_button_column) %>%
-        dplyr::select(cancel_button, dplyr::everything())
+        dplyr::select(cancel_button, dplyr::everything()) %>%
+        dplyr::bind_cols(other_button_columns, .)
```

Comment:
what if there are no other_button_columns?  should this be conditional on there actually being other columns, or in that case it still implicitly works find because other_button_columns is NULL? I would have thought this might fail in such a case

Reply:
I have to test this. I'll make sure this doesn't fail when there are no other columns.

Reply:
Confirmed that when there are no other_button_columns, the program still runs. So yes, it works implicitly because "other_button_columns" is NULL.

Reply:
Technically other_button_columns is a list of length 0, actually, but the point is that it works.